Flaky AdaptVQE test causing errors in CI

### Environment

- **Qiskit Terra version**: main @ (I'm on my phone looking at CI)
- **Python version**: 3.7
- **Operating system**: Windows


### What is happening?

I've seen a CI failure cause by a seemingly flaky cyclicity test in `AdaptVQE`, added in #7930. It doesn't seem to happen _that_ often, but the commit only merged yesterday and we've had a CI failure of it since, so it's got too high a chance.

### How can we reproduce the issue?

The CI failure was
```text
test.python.algorithms.minimum_eigensolvers.test_adapt_vqe.TestAdaptVQE.test_cyclicity
--------------------------------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "D:\a\1\s\test\python\algorithms\minimum_eigensolvers\test_adapt_vqe.py", line 119, in test_cyclicity
    self.assertEqual(res.termination_criterion, TerminationCriterion.CYCLICITY)

      File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\unittest\case.py", line 852, in assertEqual
    assertion_func(first, second, msg=msg)

      File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\unittest\case.py", line 845, in _baseAssertEqual
    raise self.failureException(msg)

    AssertionError: <Term[13 chars]ion.CONVERGED: 'Threshold converged'> != <Term[13 chars]ion.CYCLICITY: 'Aborted due to a cyclic select[23 chars]ors'>
```

### What should happen?

Tests are reliable.

### Any suggestions?

Possibly there's somewhere in that test that needs to set a random seed (`algorithm_globals.rng` or something?). I haven't looked at the test case, to know if it could possibly be a true failure, or if it's just bad luck with an optimiser stuck in a very non-optimal plateau. If the issue appears again and we can't immediately find the problem, we might consider disabling that test temporarily to avoid jamming CI during the potential RC release.

--- a/test/python/algorithms/minimum_eigensolvers/test_adapt_vqe.py
+++ b/test/python/algorithms/minimum_eigensolvers/test_adapt_vqe.py
@@ -23,6 +23,7 @@ from qiskit.circuit.library import EvolvedOperatorAnsatz
 from qiskit.opflow import PauliSumOp
 from qiskit.primitives import Estimator
 from qiskit.quantum_info import SparsePauliOp
+from qiskit.utils import algorithm_globals
 
 
 class TestAdaptVQE(QiskitAlgorithmsTestCase):
@@ -30,6 +31,7 @@ class TestAdaptVQE(QiskitAlgorithmsTestCase):
 
     def setUp(self):
         super().setUp()
+        algorithm_globals.random_seed = 42
         self.h2_op = PauliSumOp.from_list(
             [
                 ("IIII", -0.8105479805373266),
@@ -113,6 +115,7 @@ class TestAdaptVQE(QiskitAlgorithmsTestCase):
         calc = AdaptVQE(
             VQE(Estimator(), self.ansatz, self.optimizer),
             max_iterations=100,
+            threshold=1e-15,
         )
         res = calc.compute_minimum_eigenvalue(operator=self.h2_op)
 

